public/spec/features/agents_spec.rb

```diff
@@ -28,3 +25,96 @@
     expect(page).to_not have_text Pathname.new(current_path).parent.to_s
   end
 end
+
+describe 'Agent with multiple name forms', js: true do
+  before(:all) do
+    @now = Time.now.to_i
+
+
+    auth_use_date = {
+      'date_label' => 'usage',
+      'date_type_structured' => 'range',
+      'structured_date_range' => {
+        'begin_date_expression' => '2000',
+        'end_date_expression' => '2005'
+      }
+    }
+
+    alias_one_use_date = {
+      'date_label' => 'usage',
+      'date_type_structured' => 'range',
+      'structured_date_range' => {
+        'begin_date_expression' => '2006',
+        'end_date_expression' => '2010'
+      }
+    }
+
+    alias_two_use_date = {
+      'date_label' => 'usage',
+      'date_type_structured' => 'single',
+      'structured_date_single' => {
+        'date_expression' => '2011',
+        'date_role' => 'begin'
+      }
+    }
+
+    @agent_with_multiple_names = create(
+      :agent_person,
+      names: [
+        build(
+          :name_person,
+          primary_name: "Authoritative Name #{@now}",
+          authority_id: "auth_id_#{@now}",
+          is_display_name: true,
+          authorized: true,
+          use_dates: [auth_use_date]
+        ),
+        build(
+          :name_person,
+          primary_name: "Alias One #{@now}",
+          use_dates: [alias_one_use_date]
+        ),
+        build(
+          :name_person,
+          primary_name: "Alias Two #{@now}",
+          use_dates: [alias_two_use_date]
+        )
+      ],
+      publish: true
+    )
+    run_indexers
+  end
+
+  after(:all) do
+    @agent_with_multiple_names&.delete
+  end
+
+  it 'displays name usage dates correctly' do
+
+    visit('/agents')
+
+
+    click_link "Authoritative Name #{@now}"
+
+
+    expect(current_path).to match(/agents\/people\/\d+/)
+    expect(page).to have_content("Authoritative Name #{@now}")
+
+
+    within('#names_panel .card-body ul.present_list') do
```

Comment:
this selector could just be `#sidebar` which helps for readability

Reply:
Done